src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsValueSetter.cs

```diff
@@ -677,8 +690,13 @@ internal void SetDateTimeOffset(DateTimeOffset value)
             int days = utcDateTime.Subtract(DateTime.MinValue).Days;
             short offset = (short)value.Offset.TotalMinutes;
 
+#if NETCOREAPP
+            _stateObj.WriteByteArray(BitConverter.GetBytes(BinaryPrimitives.ReadInt64LittleEndian(BitConverter.GetBytes(time))), length - 5, 0); // time
+            _stateObj.WriteByteArray(BitConverter.GetBytes(BinaryPrimitives.ReadInt32LittleEndian(BitConverter.GetBytes(days))), 3, 0); // date
```

Comment:
covert, reverse, convert again? doesn't seem sensible, there should be a better way to do this.

Reply:
It's seems that BinaryPrimitive.Read...LittleEndian(byte[]) returns data type value while BinaryPrimitives.Write...LittleEndian(out byte[], datatype value) returns a bool type. there is not really a good way to deal with the help of Read and Write Functions. Although we can do something like this only for BE.
```
BitConvert.GetBytes(BinaryPrimitives.ReverseEndianness(value));
```